src/psycopmlutils/timeseriesflattener/data_integrity.py

```diff
@@ -0,0 +1,241 @@
+"""Code to generate data integrity and train/val/test drift reports."""
+from pathlib import Path
+from typing import List, Optional
+
+from deepchecks.core.suite import SuiteResult
+from deepchecks.tabular import Dataset, Suite
+from deepchecks.tabular.checks import (
+    CategoryMismatchTrainTest,
+    DatasetsSizeComparison,
+    FeatureLabelCorrelation,
+    FeatureLabelCorrelationChange,
+    IdentifierLabelCorrelation,
+    IndexTrainTestLeakage,
+    NewLabelTrainTest,
+    TrainTestLabelDrift,
+)
+from deepchecks.tabular.suites import data_integrity
+from wasabi import Printer
+
+from psycopmlutils.loaders.flattened import load_split_outcomes, load_split_predictors
+
+
+def check_feature_sets_dir(
+    path: Path,
+    splits: Optional[List[str]] = ["train", "val", "test"],
+    nrows: Optional[int] = None,
+) -> None:
+    """Runs Deepcheck data integrity and train/val/test checks for a given
+    directory containing train/val/test files. Splits indicates which data
+    splits to check. If nrows is provided, only.
+
+    the first nrows are loaded - should only be used for debugging.
+
+    The resulting reports are saved to a sub directory as .html files.
+
+    Args:
+        path (Path): Path to a directory containing train/val/test files
+        splits (List[str]): List of splits to check (train, val, test)
+        nrows (Optional[int]): Whether to only load a subset of the data.
+        Should only be used for debugging.
+    """
+    msg = Printer(timestamp=True)
+    failed_checks = {}
+
+    ## check if file splits exist before running checks
+    for split in splits:
+        file = list(path.glob(f"*{split}*.csv"))
+        if not file or len(file) > 1:
+            raise ValueError(f"{split} split not found in {path}")
+
+    out_dir = path / "deepchecks"
+    if not out_dir.exists():
+        out_dir.mkdir()
+    # create subfolder for outcome specific checks
+    outcome_checks_dir = out_dir / "outcomes"
+    if not outcome_checks_dir.exists():
+        outcome_checks_dir.mkdir()
+
+    ###################
+    #### DATA INTEGRITY
+    ###################
+
+    if "train" in splits:
+        msg.info("Running data integrity checks...")
+        # Only running data integrity checks on the training set to reduce the
+        # chance of any form of peaking at the test set
+        train_predictors = load_split_predictors(
+            path=path,
+            split="train",
+            include_id=True,
+            nrows=nrows,
+        )
+        ds = Dataset(
+            df=train_predictors,
+            index_name="dw_ek_borger",
+            datetime_name="timestamp",
+        )
+
+        # Running checks that do not require a label
+        integ_suite = data_integrity(timeout=0)  # timeout=0 removes timeout
```

Comment:
```suggestion
        integ_suite = data_integrity(timeout=900)  # timeout for each test in the suite
```

Hmm, do we want infinite time-out? Or maybe something like 15 minutes per suite?

Reply:
Sure!